minkindr_gtsam/test/test_minkindr-gtsam.cc

```diff
@@ -110,10 +109,15 @@ TEST(MinkindrGtsamTests, testTransformPoint) {
 
   EVector3 Tp = T * p;
 
-  const double fd_step = 1e-9;
-  const double tolerance = 1e-6;
-  SCOPED_TRACE("Testing Expression Jacobians.");
-  testExpressionJacobians(Tp, values, fd_step, tolerance);
+  for (int i = 0; i < N_TEST_ITERATIONS; ++i) {
+    Tval.setRandom();
+    pval.setRandom();
+    values.update(1, Tval);
+    values.update(2, pval);
```

Comment:
Didn't you try to do small random steps here? What is the distribution of setRandom here? Is it of small variance? Same applies of course to all other tests sharing this concept. I would actually do the random update below the test and start with the identity. This way you always starting with testing at exact identity - you have code that gets only activated then and these random updates never get you at exact identity. Maybe have an comment somewhere why you chose deliberate this order then.